test/testLeaderboard.py

```diff
@@ -50,144 +52,171 @@ def tearDown(self):
 
     def testAlias_add_jeeeeeeef(self):
-        self.assertFalse( jeef in self.bot.alias.keys() )
-        self.bot.addAlias(jeef, 'jeef')
+        self.assertFalse(jeef in self.bot.alias.keys())
+        self.bot.addAlias(jeef, "jeef")
         new = self.bot.getNewAlias()
-        self.assertEqual('jeef', new[jeef])
-        self.assertEqual(1+len(default_alias), len(new))
+        self.assertEqual("jeef", new[jeef])
+        self.assertEqual(1 + len(default_alias), len(new))
 
     def testAlias_add_jeeeeeeef2(self):
-        self.assertFalse( jeef in self.bot.alias.keys() )
-        self.bot.addAlias(jeef, 'jeef')
+        self.assertFalse(jeef in self.bot.alias.keys())
+        self.bot.addAlias(jeef, "jeef")
         new = self.bot.getNewAlias()
-        self.assertEqual('jeef', new[jeef])
-        self.assertEqual(1+len(default_alias), len(new))
+        self.assertEqual("jeef", new[jeef])
+        self.assertEqual(1 + len(default_alias), len(new))
         new = self.bot.getNewAlias()
-        self.assertEqual('jeef', new[jeef])
-        self.assertEqual(1+len(default_alias), len(new))
+        self.assertEqual("jeef", new[jeef])
+        self.assertEqual(1 + len(default_alias), len(new))
 
     def testChannel_add_jeeeeeeef(self):
-        self.assertFalse( jeef in self.bot.getChannels() )
-        self.bot.addChannel(jeef, 'jeef')
+        self.assertFalse(jeef in self.bot.getChannels())
+        self.bot.addChannel(jeef, "jeef")
         new = self.bot.getNewChannels()
-        self.assertEqual('jeef', new[jeef])
+        self.assertEqual("jeef", new[jeef])
         self.assertEqual(1, len(new))
 
-
     def testChannel_add_jeeeeeeef2(self):
-        self.assertFalse( jeef in self.bot.getChannels() )
-        self.bot.addChannel(jeef, 'jeef')
+        self.assertFalse(jeef in self.bot.getChannels())
+        self.bot.addChannel(jeef, "jeef")
         new = self.bot.getNewChannels()
-        self.assertEqual('jeef', new[jeef])
+        self.assertEqual("jeef", new[jeef])
         self.assertEqual(1, len(new))
         new = self.bot.getNewChannels()
         self.assertEqual(0, len(new))
 
-
-
-if __name__ == '__main__':
+    def test_get_top16(self):
+        top_16_data = self.bot.get_leaderboard_range(1, 16)
+        for region in top_16_data.keys():
+            assert len(top_16_data[region]) == 16
+            player_name_set = set()
+            rank_counter = 1
+            previous_rating = float("inf")
+            for rank, rating, name in top_16_data[region]:
+                # Assert that all players in top 16 are unique.
+                assert name not in player_name_set
+                player_name_set.add(name)
+                # Assert that ranks are descending 1->16
+                assert rank == rank_counter
+                rank_counter += 1
+                # Assert that ratings are in descending order.
+                assert rating <= previous_rating
+                previous_rating = rating
```

Comment:
Sorry about the huge diffs again. @quinnabrvau feel free to review this. I used native asserts instead of the unit test asserts, let me know if you want me to change them.